src/client/__tests__/index.spec.js

```diff
@@ -114,13 +114,29 @@ describe( 'ApiClient', () => {
 
 		const component = () => {};
 
+		it( 'should set and clear component requirements', () => {
+			const apiClient = new ApiClient( api, '123' );
+
+			apiClient.setComponentData( component, ( selectors ) => {
+				selectors.getThing( { freshness: 60 * SECOND }, 1 );
+			} );
+			expect( apiClient.requirementsByComponent.size ).toEqual( 1 );
+			expect( apiClient.requirementsByComponent.get( component ) ).toEqual( [
+				{ freshness: 60 * SECOND, endpoint: [ 'things', 1 ] },
+			] );
+
+			apiClient.setComponentData( component, null );
+			expect( apiClient.requirementsByComponent.size ).toEqual( 0 );
+		} );
+
 		it( 'should select data for component from last state set', () => {
 			const apiClient = new ApiClient( api, '123' );
 			apiClient.setState( thing1ClientState );
 
 			apiClient.setComponentData( component, ( selectors ) => {
 				expect( selectors.getThing( {}, 1 ) ).toBe( thing1 );
 			} );
+			apiClient.setComponentData( component, null );
```

Comment:
Oh, yeah it might, I didn't think to do that since it isn't necessary for all the tests. But I suppose it wouldn't hurt anything to do that after each test.